ADR/0026-specifying-ocp-targets-for-fbc.md

```diff
@@ -0,0 +1,72 @@
+# 26. Specifying OCP targets for File-based Catalogs
+
+* Date 2023-06-08
+
+## Status
+
+Accepted
+
+## Context
+
+One of the supported component types within AppStudio are[File-based Catalogs (FBC)].
+These catalogs can either be used in isolation with a version of `opm` packaged in the container
+itself or in conjunction with other catalog configurations via a service such as
+[IIB Image Builder]. Red Hat OpenShift Container Platform (OCP) is one example of a platform that
+leverages FBCs for defining the operator graphs. In order to enable operator support to vary on a
+version-by-version basis, Red Hat maintains one catalog per OpenShift version.
+
+In order to support being able to target FBC components to specific versions of Red Hat OpenShift,
+AppStudio needs to be able to keep track of the specific targeted version. In addition to the concerns
+around releasing FBC components to OpenShift, the version of `opm` used by each version of OpenShift
+may differ, so the AppStudio integration process will need to ensure that tests are run using an appropriate
+binary version.
+
+## Decision
+
+All FBC components intending to be released to OCP will be built using a OCP-specific parent image containing
+the target version number as a tag. This will result in a `FROM` instruction like
+
+```
+# The base image is expected to contain
+# /bin/opm (with a serve subcommand) and /bin/grpc_health_probe
+FROM registry.redhat.io/openshift4/ose-operator-registry:v4.12
```

Comment:
is v4.12 added by the build process?
or does the build process use the tag from the line to determine the version?

Reply:
No, this is from the Dockerfile itself. Buildah uses the information to populate the annotation. I could not find a specific documentation for this feature, but in the [v1.30.0 release notes](https://buildah.io/releases/2023/04/21/Buildah-version-v1.30.0.html), it indicates

> OCI images produced using multi-stage builds, where the final stage is based on an earlier stage, will no longer include an “org.opencontainers.image.base.name” annotation for the base of that earlier stage combined with an “org.opencontainers.image.base.digest” annotation which corresponds to the image produced by that earlier stage, which are two different images. [#4618](https://github.com/containers/buildah/pull/4618)

Indicating that the image references will map only to the final stage of an image build.

Reply:
That makes sense. Can you please add this statement then:

> Buildah uses the information to populate the annotation